src/openms/source/FORMAT/ConsensusXMLFile.cpp

```diff
@@ -342,22 +342,24 @@ namespace OpenMS
       prot_id_.setSearchEngine(attributeAsString_(attributes, "search_engine"));
       prot_id_.setSearchEngineVersion(attributeAsString_(attributes, "search_engine_version"));
       prot_id_.setDateTime(DateTime::fromString(String(attributeAsString_(attributes, "date")).toQString(), "yyyy-MM-ddThh:mm:ss"));
-      //set identifier
-      String identifier = prot_id_.getSearchEngine() + '_' + attributeAsString_(attributes, "date");
+      // set identifier
+      // always generate a unique id to link a ProteinIdentification and the corresponding PeptideIdentifications
+      // , since any FeatureLinker might just carelessly concatenate PepIDs from different FeatureMaps.
+      // If these FeatureMaps have identical identifiers (SearchEngine time + type match exactly), then ALL PepIDs would be falsely attributed
+      // to a single ProtID...
+
       String id = attributeAsString_(attributes, "id");
+      while (true)
+      { // loop until the identifier is unique (should be on the first iteration -- very(!) unlikely it will not be unique)
```

Comment:
why do you need the loop? the uid is 64 bit which gives a space of 2 E19 which should always be unique ? Are you concerned about multiple threads?

Reply:
PS: also, why dou still need the search engine and the date - shouldnt the UID be enough?

Reply:
PS: the probability of any two ids not being unique with 64 bits and assuming that we have a hundred thousand to a few million spectra is still 1e-9 to 1e-6: https://en.wikipedia.org/wiki/Birthday_attack#Mathematics

Reply:
the loop is just there for MurphysLaw... thats all.
In general, there will be a single ProtID, in rare cases a handful. But why test our luck?
This is still not avoiding the problem of merging protIDs at a later stage.. but its a start.